sbi/inference/posteriors/direct_posterior.py

```diff
@@ -288,10 +269,11 @@ def sample(
         sample_shape: Shape = torch.Size(),
         x: Optional[Tensor] = None,
         show_progress_bars: bool = True,
-        sample_with_mcmc: Optional[bool] = None,
+        sample_with: Optional[str] = None,
         mcmc_method: Optional[str] = None,
         mcmc_parameters: Optional[Dict[str, Any]] = None,
         rejection_sampling_parameters: Optional[Dict[str, Any]] = None,
+        sample_with_mcmc: Optional[bool] = None,
```

Comment:
Nope, you are totally write. I'll add a note about this.

One small thing: in case the user passes `rejection_sample_parameters={"proposal": self.prior}` then we would do actual rejection sampling for SNPE.